### ISSUE TITLE
Does not passes CTS profile match in android 12

### ISSUE THREAD

**Issue by crossdefalt** [kdrag0n/safetynet-fix#187]
After successfully installing latest module in magisk these are the output of SafetyNet test 
Safety Net Request success
Response signature validation success
Basic Integrity success
CTS profile match fail

**Comment by Attorelle** [comment]
Try to use "prop spoofer" to change yours fingerprint, etc.

**Comment by brightdroid** [comment]
I've the same problem with latest OnePlus 8T version KB2003_11_C.20

**Comment by ruszok** [comment]
> I've the same problem with latest OnePlus 8T version KB2003_11_C.20

Same.
![Screenshot_2022-05-30-21-29-33-76_9492aa3750dca76abb7c25b39a5f1e8e](https://user-images.githubusercontent.com/68990273/171057918-7786465b-97f2-49a7-bd4d-cfd2236936fa.jpg)
![Screenshot_2022-05-30-22-37-13-05_4aae5e5c7232b4014d16720d2baa6855](https://user-images.githubusercontent.com/68990273/171057983-8506fa84-81dc-4d68-ae54-81867554a312.jpg)

**Comment by idanny76** [comment]
Any fix yet?

**Comment by crossdefalt** [comment]
![Uploading Screenshot_20220602-203618_SafetyNet Test.png…]()

**Comment by qriozum** [comment]
@ruszok

Colleague @rori999 for the OP8PRO model checked Fingerprint from the "10" and "11" versions and they did not work, despite the change in "Magiskhide" still did not undergo the test "CTS Profiles Matched".

On your screenshot, what is the module marked as the third, is it necessary?

After changing Fingerprint in "Magiskhide", will the system be updated by the official channel via OTA?

**Comment by ruszok** [comment]
3 it's necessary for fingerprint reader.

**Comment by qriozum** [comment]
> 1, 2 - dla safetynet (po tym safetynet działa). 3 - jest niezbędny dla czytnika linii papilarnych.

@Ruszok - for a fingerprint scanner, "SafetyNet -Fix" is enough, but in the V2.3.0 version.

Can you answer my question related to OTA updates after changing Fingerprint in "Magiskhide"?

**Comment by ruszok** [comment]
In c20 fingerprint works after safetynet 2.3.0 but safetynet fail.
If change fingerprint by magiskhide props safetynet passed but fingerprint reader doesn't work. So must install ColorOS module.
I can't update OTA incremental after magisk. OTA full works - if there is full package.

**Comment by qriozum** [comment]
> W c20 czytnik linii papilarnych działa po safetynet 2.3.0 (lub 2.2.1 z modem), ale safetynet zawodzi. Jeśli zmienisz odcisk palca urządzenia przez magiskhide props safetynet przeszedł, ale czytnik linii papilarnych nie działa. Więc musi zainstalować moduł ColorOS - wtedy czytnik linii papilarnych działa dobrze.
> 
> Nie mogę zaktualizować przyrostowego OTA po magisk. OTA działa w całości - jeśli jest pełny pakiet.

@ruszok 
I understand now, thanks for the clarification.

You're right, after root (via Magisk) incremental OTAs never worked, you have to wait for the full OTA (full ROM) in that case.

**Comment by idanny76** [comment]
> > In c20 fingerprint reader works after safetynet 2.3.0 (or 2.2.1 with mod) but safetynet fail.
> If change device fingerprint by magiskhide props safetynet passed but fingerprint reader doesn't work. So must install ColorOS module - then fingerprint reader works fine.
> 
> > I can't update OTA incremental after magisk. OTA full works - if there is full package.
> 
> @ruszok 
> I understand now, thanks for the clarification.
> 
> You're right, after root (via Magisk) incremental OTAs never worked, you have to wait for the full OTA (full ROM) in that case.

I have updated incremental with magisk for a while now. What I do is in Magisk.... I select Uninstall Magisk and Restore Images. After that...... I go to the incremental OTA update and install. When the installation finishes and ask for a reboot, I go back to Magisk, select install on inactive slot and reboot.

**Comment by ruszok** [comment]
Right. It is possible like that. But I don't remember - magisk settings are restored after that?

**Comment by idanny76** [comment]
> Right. It is possible like that. But I don't remember - magisk settings are restored after that?

Yes, everything gets restored with magisk. You do not need to reinstall or configure any settings. Everything in Magisk is saved and transferred to the new slot. It works exactly as if you were installing the full OTA and then going into magisk and installing in inactive slot. The only extra step for incremental updates is restoring boot images. Because for a full OTA doesn't care about boot img, but incremental does.

**Comment by ruszok** [comment]
Ok. So I will try when will be OTA incremental.

If install full package I also need to reinstall magisk (option: after OTA). Without that lost magisk - do it's same in both cases right? Or I misunderstand something?

**Comment by idanny76** [comment]
> Ok. So I will try when will be OTA incremental.
> 
> If install full package I also need to reinstall magisk (option: after OTA). Without that lost magisk - do it's same in both cases right? Or I misunderstand something?

When you install full OTA, you still need to go into magisk before reboot to install in inactive slot. If you don't, you have to obtain boot.img and root again. When is incremental.... If you don't uninstall Magisk to restore boot img... It will fail. When I say uninstall Magisk is doing a restore boot images and not a complete uninstall. Once you restore images, it will allow you to install incremental and then before reboot install magisk to inactive slot.

**Comment by ruszok** [comment]
Understood. Will try. Thanks.

**Comment by alfredonodo** [comment]
> 1, 2 - for safetynet (after that safetynet works). 3 - it's necessary for fingerprint reader.

I tried, but it doesn't work for me. I only have microG instead of Google Services. Could this be the problem? If so, is there a way around it?
Thanks

**Comment by rori999** [comment]
Hi Guys, I tried to update my Oneplus 8 Pro to latest C33 version of Android 12. But couldn't. Using this https://www.reddit.com/r/OnePlus8Pro/comments/w8db92/instructions_on_how_to_install_full_ota_update/?utm_medium=android_app&utm_source=share) I could update. When I updated I turned off all suggestions and still have Google Pay working. I guess the issue is fixed with latest Oneplus firmware.

**Comment by Vladi69** [comment]
Great solution! Finally Gpay working fine (again) on my OP8T with A12

**Comment by Zibri** [comment]
I have basic integrity pass and cts fail with original stock rom and only the bootloader unlocked. (on YASNAC)
On play integrity checker (which uses the new API) I have red crosses on meet_device_integrity and meets strong integrity but a green on meets basic integrity.
On a clean phone wiped and factory restored.
(but with unlocked bootloader)

**Comment by SsNiPeR1** [comment]
any android 13 workaround? cus i did all the following previously on a11 & a12 and everything worked, but android 13 for some reason does not work

**Comment by Zibri** [comment]
with the latest magisk delta it passes cts but only BASIC...
I really wonder if there is a way to make it pass completely..

**Comment by yyehhyyaa** [comment]
![Screenshot_20230821-152458_Yet Another SafetyNet Attestation Checker](https://github.com/kdrag0n/safetynet-fix/assets/142803865/426e8320-37d0-4f61-8ea2-68e408ccf618)
Any solution?

**Comment by SsNiPeR1** [comment]
guys google the safetynet fix MOD, it works for me

**Comment by alfredonodo** [comment]
> ![Screenshot_20230821-152458_Yet Another SafetyNet Attestation Checker](https://user-images.githubusercontent.com/142803865/262022289-426e8320-37d0-4f61-8ea2-68e408ccf618.jpg) Any solution?
You can try to follow [this](https://community.e.foundation/t/safetynet-on-e-os-community-dev-devices-let-it-in-or-not/46852/15?u=alfredino).

**Comment by idanny76** [comment]
> > ![Screenshot_20230821-152458_Yet Another SafetyNet Attestation Checker](https://user-images.githubusercontent.com/142803865/262022289-426e8320-37d0-4f61-8ea2-68e408ccf618.jpg) Any solution?
> You can try to follow [this](https://community.e.foundation/t/safetynet-on-e-os-community-dev-devices-let-it-in-or-not/46852/15?u=alfredino).
> 

I am using this safetynet mod and it's working for me.
https://github.com/Displax/safetynet-fix
